### PR TITLE
Feat 2.1.x/ab#70943 allow multiple fields update when editing role permissions

### PR DESCRIPTION
**Pull request by NathanHGit**
# Description

Added ability to update permissions for multiple fields at once.

## Ticket
Ticket: [70943 - Allow multiple fields to be updated at same time when editing role permissions](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/70943)
Frontend [PR #1672](https://github.com/ReliefApplications/oort-frontend/pull/1672)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I tried the new front-end functionality to modify permissions of several fields and checked if the modifications were relevant.

# Checklist:

( * == Mandatory ) 

- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Maximo-reliefapps — COMMENTED** [review]
Everything working as described, using the correct back branch and all good on the code.
Only issue I see is that it can be difficult to relate the permissions with the icons, a possible solution is to add the icons along the descriptions on the modal.

**Review by estelafs — COMMENTED** [review]
Removed incorrect `if `statement that avoided the checkFieldPermission() to check if the resource has the necessary permissions that allowed the field to have its own permissions. The `if` check if the field's canSee permission was being updated, which doesn't give permission for the field without first checking if the resource has the canSeeRecords or canCreateRecords permission. 

But the incorrect return the `if ` was fixing is better fixed now inside the checkFieldPermission()  with the new updatingCanSee param to see if the field permission canSee is being updated at the same time as the canUpdate permission

File: src/schema/mutation/editResource.mutation.ts [hunk 1]
```diff
@@ -666,36 +690,78 @@ export default {
...
+            if (obj.add) {
+              obj.add.fields.forEach((field) => {
+                checkFieldPermission(
+                  context,
+                  get(resource, 'permissions'),
+                  allResourceFields,
+                  field,
+                  obj.add.role,
+                  permission,
+                  !!permissions.canSee.add
```
Comment (RenzoPrats) [inline src/schema/mutation/editResource.mutation.ts:742]: This is fine, a little suggestion that could be more readable
```suggestion
                  ('canSee' in permissions && 'add' in permissions.canSee)
```

**Review by RenzoPrats — COMMENTED** [review]
Everything is working as intended.

**Review by RenzoPrats — APPROVED** [review]

**Pull request closed without merging**